src/main.ts

```diff
@@ -1,16 +1,70 @@
+import { promises as fs } from "fs";
+import path from "path";
 import * as core from "@actions/core";
-import { wait } from "./wait";
+import fetch, { Headers } from "node-fetch";
+
+interface Config {
+  requestToken: string;
+  requestUrl: string;
+  awsRoleArn: string;
+  awsRegion: string;
+  gitHubEnvFile: string;
+}
+
+function getEnvString(name: string): string {
+  const value = process.env[name];
+  if (!value) {
+    throw new Error(`${name} env var not found`);
+  }
+  return value;
+}
+
+async function fetchWebIdentityToken(config: Config): Promise<string> {
+  const headers = new Headers({ Authorization: `bearer ${config.requestToken}` });
+  const response = await fetch(`${config.requestUrl}&audience=sigstore`, {
+    headers,
+  });
+  const responseJson = (await response.json()) as Record<string, unknown>;
+  return responseJson.value as string;
+}
 
 async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput("milliseconds");
-    core.debug(`Waiting ${ms} milliseconds ...`); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
+    const config: Config = {
+      requestToken: getEnvString(`ACTIONS_ID_TOKEN_REQUEST_TOKEN`),
+      requestUrl: getEnvString("ACTIONS_ID_TOKEN_REQUEST_URL"),
+      awsRoleArn: core.getInput("awsRoleToAssume"),
+      awsRegion: core.getInput("awsRegion"),
+      gitHubEnvFile: getEnvString("GITHUB_ENV"),
+    };
+
+    core.debug("*** START config ***");
+    core.debug(JSON.stringify(config));
```

Comment:
IIUC debug logs will be [hidden by default](https://github.com/actions/toolkit/tree/main/packages/core#logging), but users can [choose to enable these log lines manually](https://github.com/actions/toolkit/blob/main/docs/action-debugging.md#how-to-access-step-debug-logs). Should we remove this debug logging before publishing to avoid allowing people to accidentally leak the `awsRoleToAssume` or `ACTIONS_ID_TOKEN_REQUEST_TOKEN`?

Reply:
I think any token or secret-like strings were automatically redacted. We added these to help debugging, but we've proven the process works now so I think we can delete these lines.